components/fastly_backends/main.tf

```diff
@@ -15,48 +21,76 @@ locals {
   }
 }
 
-resource "fastly_service_v1" "backends-qa" {
-  name          = "Terraform: Backends (QA)"
+resource "fastly_service_v1" "backends" {
+  name          = "Terraform: Backends"
   force_destroy = true
 
-  domain {
-    name = var.northstar_domain
-  }
+  # Configure backend, domain, and condition for each app:
+  dynamic "domain" {
+    for_each = var.backends
 
-  domain {
-    name = var.rogue_domain
+    content {
+      name = domain.value.domain
+    }
   }
 
-  condition {
-    type      = "REQUEST"
-    name      = "backend-northstar-qa"
-    statement = "req.http.host == \"${var.northstar_domain}\""
+  dynamic "condition" {
+    for_each = var.backends
+
+    content {
+      type      = "RESPONSE"
+      name      = "response-${condition.value.name}"
+      statement = "req.http.host == \"${condition.value.domain}\""
+    }
   }
 
-  condition {
-    type      = "RESPONSE"
-    name      = "response-northstar-qa"
-    statement = "req.http.host == \"${var.northstar_domain}\""
+  dynamic "condition" {
+    for_each = var.backends
+
+    content {
+      type      = "REQUEST"
+      name      = "backend-${condition.value.name}"
+      statement = "req.http.host == \"${condition.value.domain}\""
+    }
   }
 
-  condition {
-    type      = "REQUEST"
-    name      = "backend-rogue-qa"
-    statement = "req.http.host == \"${var.rogue_domain}\""
+  dynamic "backend" {
+    for_each = var.backends
+
+    content {
+      address           = backend.value.backend
+      name              = backend.value.name
+      request_condition = "backend-${backend.value.name}" # defined above.
+      port              = 443
+
+      # We use shielding to increase chances that we get cache hits. We don't rely
+      # on Fastly for load balancing (since that is handled by Heroku's router).
+      shield           = "bwi-va-us"
+      auto_loadbalance = false
+    }
   }
 
-  condition {
-    type      = "RESPONSE"
-    name      = "response-rogue-qa"
-    statement = "req.http.host == \"${var.rogue_domain}\""
+  # Fix issue with geolocation when shielding is enabled:
+  snippet {
+    name    = "Fix Geolocation With Shielding On"
+    type    = "recv"
+    content = file("${path.module}/recv_geolocation.vcl")
   }
 
+  # Configure 'robots.txt' global deny for backends:
   condition {
     type      = "REQUEST"
     name      = "path-robots"
     statement = "req.url.basename == \"robots.txt\""
   }
 
+  response_object {
+    name              = "robots.txt deny"
+    content           = file("${path.module}/deny-robots.txt")
+    request_condition = "path-robots"
+  }
```

Comment:
Likewise, moved the `response_object` here up so it's next to the condition it relies on.